### PR TITLE
Heading component

### PR DESCRIPTION
**Pull request by auqibyousuf**

### CODE REVIEW

**Comment by UmmarHamid** [comment]
@auqibyousuf This PR has previous commit as well.
It should not have previous work right?

File: app/components/Heading/Heading.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const headingClasses = classnames({
+    "text-[20px] leading-[24px]": variant == "20",
+    "text-[22px] leading-[28px]": variant == "22",
+    "text-[24px] leading-[30px]": variant == "24",
+    "text-[32px] leading-[40px]": variant == "32",
+    "text-[36px] leading-[44px]": variant == "36",
+    "text-[48px] leading-[48px]": variant == "48",
+  });
+  return (
+    <DynamicTag className={`${headingClasses} ${extraClasses}`}>
```
Comment (UmmarHamid) [inline app/components/Heading/Heading.tsx]: Can we name this as DynamicHeading instead?

File: app/components/Heading/Heading.stories.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  tags: ["autodocs"],
+  parameters: {
+    layout: "fullscreen",
+  },
+} satisfies Meta<typeof Text>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const HeadingVariant: Story = {
```
Comment (UmmarHamid) [inline app/components/Heading/Heading.stories.ts]: Just name is Heading 20,
Same for Text

**Review by UmmarHamid — CHANGES_REQUESTED** [review]
The dynamic tag approach is great !.
Small feedback

File: app/components/Heading/Heading.stories.ts (hunk 2 above)
Comment (auqibyousuf) [reply app/components/Heading/Heading.stories.ts]: ok

**Comment by auqibyousuf** [comment]
This dont have previous work created a branch f4om feature branch thats why showing previous commit

File: app/components/Heading/Heading.tsx (hunk 1 above)
Comment (auqibyousuf) [reply app/components/Heading/Heading.tsx]: ok

**Comment by UmmarHamid** [comment]
> This dont have previous work created a branch f4om feature branch thats why showing previous commit

Correct, but we don't want the same work pushed twice.
You have 2 options:
1- Either create branch off develop and add only required commit.
2- change the target branch to the previous text component branch, this way it will be merge to that branch instead and have only one commit

**Comment by auqibyousuf** [comment]
Check now

**Comment by auqibyousuf** [comment]
@UmmarHamid

**Comment by UmmarHamid** [comment]
> Check now
I don't see the implementation change yet.

**Comment by UmmarHamid** [comment]
> > Check now
> > I don't see the implementation change yet.

Also you can add new commits no need to reset and force push

File: app/components/Heading/Heading.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import classnames from "classnames";
+
+const Heading = ({
+  text,
+  variant,
+  headingElement,
+  extraClasses,
+}: HeadingTypes) => {
+  const DynamicHeading = `h${headingElement}` as keyof JSX.IntrinsicElements;
+
```
Comment (UmmarHamid) [inline app/components/Heading/Heading.tsx]:
```suggestion
  let DynamicHeading = `h1` as keyof JSX.IntrinsicElements;
  switch(variant){
    case '48':
      DynamicHeading = 'h1'
      break;
      case '36':
      DynamicHeading = 'h2'
      break;
      ......
  }

```

**Review by UmmarHamid — CHANGES_REQUESTED** [review]

**Comment by auqibyousuf** [comment]
Done

**Review by UmmarHamid — APPROVED** [review]
Looks good

**Pull request merged**